### PR TITLE
Implement creating and listing child campaigns

### PR DESCRIPTION
**Pull request by nkly**
The PR is blocked on https://github.com/advancedtelematic/libats/pull/91 which is needed for returning a proper error when non-existing `parent_campaign_id` is given during campaign creation.

### CODE REVIEW

**Comment by houcros** [comment]
@nkly it seems it won't be so easy to get that PR into libats, so I pushed one commit so we can move on with this topic for now.

**Comment by nkly** [comment]
@houcros Thank you. Sounds fine to me. Maybe you can also take a look at #93? We can merge #93 first, get a green build in this PR and merge it afterwards. WDYT?

**Comment by houcros** [comment]
Yes, that makes sense to me. #93 looks fine to me overall. But I think my approval alone wouldn't be enough to merge it, would need one of the leads (?).

**Comment by nkly** [comment]
Okay, I see. Thanks for taking a look at it anyway!

File: src/main/scala/com/advancedtelematic/campaigner/db/Repository.scala [hunk 1]
```diff
@@ -97,7 +99,7 @@ protected [db] class DeviceUpdateRepository()(implicit db: Database, ec: Executi
       .map(_.status)
       .update(status)
       .flatMap {
-        case 0 => DBIO.failed(Errors.DeviceNotScheduled)
+        case 0 => DBIO.failed(DeviceNotScheduled)
```
Comment (jerrytrieu) [inline src/main/scala/com/advancedtelematic/campaigner/db/Repository.scala]: Let's avoid prettying/formatting/whitespace changes in `feat`/`bugfix` PRs. Better to have separate `refactor` branches.

File: src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala [hunk 2]
```diff
@@ -90,13 +93,15 @@ object DataType {
...
     updatedAt: Instant,
+    parentCampaignId: Option[CampaignId],
+    childCampaignIds: Set[CampaignId],
     groups: Set[GroupId],
     metadata: Seq[CreateCampaignMetadata],
     autoAccept: Boolean
   )
 
   object GetCampaign {
-    def apply(c: Campaign, groups: Set[GroupId], metadata: Seq[CampaignMetadata]): GetCampaign =
```
Comment (houcros) [inline src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala]: I was thinking, do we actually want to send the child campaigns? I don't think we'll need them. Only the child campaign stats when requesting the stats of the parent.

**Comment by houcros** [comment]
Please let's have a look at #95 before we proceed to merge this.

File: src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala (hunk 2 above)
Comment (nkly) [reply src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala]: Unfortunately I don't have enough knowledge to address this, but I think we might need them to actually get the stats for them. WDYT @jerrytrieu?

**Comment by nkly** [comment]
Hey @houcros looks like you've overwritten the changes that prevented child campaigns from appearing in the main list of campaigns: https://github.com/advancedtelematic/campaigner/commit/b22fc782eafc4a8c77884bb666d2ce8ff2f0789e. Is it not required anymore?

**Comment by houcros** [comment]
Sorry, seems it got lost yesterday while rebasing onto latest master 😕. We should add that back.

**Comment by nkly** [comment]
No worries, that shouldn't be hard to fix. I trust you on this :) 
I believe in the future it would be nice to use additional pull requests for such changes so that they are visible and we don't step on each other's toes. WDYT?

**Comment by houcros** [comment]
Opened #97.

**Pull request merged**